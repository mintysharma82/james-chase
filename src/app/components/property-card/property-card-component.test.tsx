import { fireEvent, render } from "@testing-library/react";
import SearchCard from "./property-card.component";
import React from "react";
import { screen } from "@testing-library/dom";

describe("Component > Property Card", () => {
  it("renders with rating as expected", () => {
    const mockHolidayData = {
      content: {
        hotelFacilities: ["Restaurant", "Swimming Pool"],
        starRating: "5",
        atAGlance: [],
      },
    };
    const { container } = render(
      <SearchCard hotel={mockHolidayData} pricePerPerson={500} />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders without rating as expected", () => {
    const mockHolidayData = {
      content: {
        hotelFacilities: [],
        atAGlance: [],
      },
    };
    const { container } = render(
      <SearchCard hotel={mockHolidayData} pricePerPerson={20} />
    );
    expect(container).toMatchSnapshot();
  });
});
