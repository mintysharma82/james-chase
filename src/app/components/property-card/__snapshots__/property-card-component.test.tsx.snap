// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > Property Card renders with rating as expected 1`] = `
<div>
  <article
    class="sc-11f3f108-1 ksixbF"
  >
    <div
      class="sc-11f3f108-2 iqfgeT"
    >
      <h2 />
      <p
        class="sc-11f3f108-3 eTIoTn"
      />
      <p>
        <span
          class="rating-5star rating-star"
        />
      </p>
      <button
        class="sc-11f3f108-0 kVrQDz"
        data-testid="show-details"
      >
        Show Details
      </button>
    </div>
    <div
      class="sc-11f3f108-6 ehxttO"
    >
      <div
        class="sc-11f3f108-5 fyNlvK"
      >
        £
        500
        <span>
          pp
        </span>
      </div>
    </div>
  </article>
</div>
`;

exports[`Component > Property Card renders without rating as expected 1`] = `
<div>
  <article
    class="sc-11f3f108-1 ksixbF"
  >
    <div
      class="sc-11f3f108-2 iqfgeT"
    >
      <h2 />
      <p
        class="sc-11f3f108-3 eTIoTn"
      />
      <p />
      <button
        class="sc-11f3f108-0 kVrQDz"
        data-testid="show-details"
      >
        Show Details
      </button>
    </div>
    <div
      class="sc-11f3f108-6 ehxttO"
    >
      <div
        class="sc-11f3f108-5 fyNlvK"
      >
        £
        20
        <span>
          pp
        </span>
      </div>
    </div>
  </article>
</div>
`;

exports[`Component > Property Card should show details when details button is clicked 1`] = `
<div>
  <article
    class="sc-11f3f108-1 ksixbF"
  >
    <div
      class="sc-11f3f108-2 iqfgeT"
    >
      <h2 />
      <p
        class="sc-11f3f108-3 eTIoTn"
      />
      <p />
      <button
        class="sc-11f3f108-0 kVrQDz"
        data-testid="show-details"
      >
        Show Details
      </button>
      <div>
        <p />
        <ul
          class="sc-11f3f108-4 eBwZep"
        />
        <h3>
          Amenities
        </h3>
        <ul
          class="sc-11f3f108-4 eBwZep"
        />
      </div>
    </div>
    <div
      class="sc-11f3f108-6 ehxttO"
    >
      <div
        class="sc-11f3f108-5 fyNlvK"
      >
        £
        20
        <span>
          pp
        </span>
      </div>
    </div>
  </article>
</div>
`;
